libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserUtils.java

```diff
@@ -36,6 +39,45 @@ public class PolicyParserUtils {
 
     private static final Logger logger = LogManager.getLogger(PolicyParserUtils.class);
 
+    public static List<Scope> mergeScopes(List<Scope> mainScopes, List<Scope> additionalScopes) {
+        var result = new ArrayList<Scope>();
+        var additionalScopesMap = additionalScopes.stream().collect(Collectors.toMap(Scope::moduleName, Scope::entitlements));
+        for (var mainScope : mainScopes) {
+            List<Entitlement> additionalEntitlements = additionalScopesMap.remove(mainScope.moduleName());
+            if (additionalEntitlements != null) {
+                result.add(new Scope(mainScope.moduleName(), mergeEntitlements(mainScope.entitlements(), additionalEntitlements)));
+            } else {
+                result.add(mainScope);
+            }
+        }
+
+        for (var remainingEntry : additionalScopesMap.entrySet()) {
+            result.add(new Scope(remainingEntry.getKey(), remainingEntry.getValue()));
+        }
+        return result;
+    }
+
+    static List<Entitlement> mergeEntitlements(List<Entitlement> a, List<Entitlement> b) {
+        Map<Class<? extends Entitlement>, List<Entitlement>> allEntitlements = Stream.concat(a.stream(), b.stream())
+            .collect(Collectors.groupingBy(Entitlement::getClass));
+
+        List<Entitlement> result = new ArrayList<>();
+        for (var entitlements : allEntitlements.entrySet()) {
+            var entitlementClass = entitlements.getKey();
+            if (entitlementClass.equals(FilesEntitlement.class)) {
+                var filesData = entitlements.getValue().stream().flatMap(entitlement -> {
+                    FilesEntitlement filesEntitlement = (FilesEntitlement) entitlement;
+                    return filesEntitlement.filesData().stream();
+                }).filter(x -> x.platform().isCurrent()).distinct();
+
+                result.add(new FilesEntitlement(filesData.toList()));
+            } else {
+                result.add(entitlements.getValue().get(0));
```

Comment:
Do we need something here for `WriteSystemPropertiesEntitlement`?

Reply:
I'm adding it, and realized the danger of missing this for new entitlements.
I don't  really like this, but what about adding `merge` to Entitlement?
```
public interface Entitlement {
    Entitlement merge(Entitlement other);
}
```
(no default, so you are forced to implement it?)

Reply:
I ended up adding a function with a switch expression and a default. @all let me know how you feel about it.

Reply:
Works for me. Data-oriented programming FTW.